src/tests/managed_array_tests.cpp

```diff
@@ -410,3 +410,54 @@ CUDA_TEST(ManagedArray, Move)
   array.free();
 }
 #endif // defined(CHAI_ENABLE_CUDA)
+
+
+TEST(ManagedArray, DeepCopy) {
+  chai::ManagedArray<float> array(10);
+  ASSERT_EQ(array.size(), 10u);
+
+  forall(sequential(), 0, 10, [=] (int i) {
+    array[i] = i;
+  });
+
+  chai::ManagedArray<float> copy = chai::deepCopy(array);
+  ASSERT_EQ(copy.size(), 10u);
+
+  forall(sequential(), 0, 10, [=] (int i) {
+    array[i] = 3.14159 * i;
+  });
+
+  forall(sequential(), 0, 10, [=] (int i) {
+    ASSERT_EQ(copy[i] = i);
+  });
+
+  array.free();
+  copy.free();
+}
+
+#if defined(CHAI_ENABLE_CUDA)
+CUDA_TEST(ManagedArray, DeviceDeepCopy)
+{
+  chai::ManagedArray<float> array(10, chai::GPU);
+  ASSERT_EQ(array.size(), 10u);
+
+  forall(cuda(), 0, 10, [=] __device__ (int i) {
+      array[i] = i;
+  });
+
+  chai::ManagedArray<float> copy = chai::deepCopy(array);
```

Comment:
@davidbeckingsale So my original intent with the deepCopy function was to only copy the data in the active region. Ie the `copy` array would only have a device allocation. But by nature of using the copy constructor here the data will get copied back to the host correct?

Reply:
Actually now that I look at it a little more the `operator=(ManagedArray&)` isn't provided so I'm guessing this is doing an automatic conversion to `T*` then using the `ManagedArray(T*)` constructor. What do you think the best way to go about this is?

Reply:
Nevermind, I thought providing the `operator=(std::nullptr_t)` would disable the compiler generated assignment operator, but I guess not.

Reply:
So this is working as you expect?

Reply:
Yup!